### PR TITLE
chore: improve chapter 13-14-16

### PR DESCRIPTION
**Pull request by TAdev0**
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/780)

### CODE REVIEW

File: src/ch16-01-optimizing-storage-costs.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Intuitively, several integers can be combined into a single integer if the size
 
 But, to do that, we need some bitwise operators:
 
-- multiplying or dividing an integer by a power of 2, shifts the integer value to the left or to the right respectively
+- multiplying or dividing an integer by a power of 2, shifts the integer value to the left or to the right respectively.
```
Comment (enitrat) [inline src/ch16-01-optimizing-storage-costs.md:23]: not full sentences, no full stop

Comment (enitrat) [inline src/ch16-01-optimizing-storage-costs.md:23]:
```suggestion
- multiplying or dividing an integer by a power of 2 shifts the integer value to the left or to the right respectively
```

File: src/ch16-01-optimizing-storage-costs.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ But, to do that, we need some bitwise operators:
     <span class="caption">Shifting to the left or to the right an integer value</span>
 </div>
 
-- applying a mask (`AND` operator) on an integer value isolates some bits of this integer
+- applying a mask (`AND` operator) on an integer value isolates some bits of this integer.
```
Comment (enitrat) [inline src/ch16-01-optimizing-storage-costs.md:32]:
```suggestion
- applying a mask (`AND` operator) on an integer value isolates some bits of this integer
```

File: src/ch16-02-00-composability-and-components.md [hunk 3]
```diff
@@ -200,9 +202,11 @@ to have.
 Developers can focus on their core contract logic while relying on battle-tested
 and audited components for everything else.
 
-Components can even [depend](./ch16-02-02-component-dependencies.md) on other components by restricting the
+Components can even [depend][component dependendcies] on other components by restricting the
```
Comment (enitrat) [inline src/ch16-02-00-composability-and-components.md:205]:
```suggestion
Components can even [depend][component dependencies] on other components by restricting the
```

File: src/ch16-02-00-composability-and-components.md [hunk 4]
```diff
@@ -200,9 +202,11 @@ to have.
...
 `TContractstate` they're generic on to implement the trait of another component.
-Before we dive into this mechanism, let's first look at [how components work under the hood](./ch16-02-01-under-the-hood.md).
+Before we dive into this mechanism, let's first look at [how components work under the hood][components inner working].
 
 
-[OpenZeppelin Cairo Contracts]: https://github.com/OpenZeppelin/cairo-contracts
\ No newline at end of file
+[OpenZeppelin Cairo Contracts]: https://github.com/OpenZeppelin/cairo-contracts
+[component dependendcies]: ./ch16-02-02-component-dependencies.md
```
Comment (enitrat) [inline src/ch16-02-00-composability-and-components.md:211]:
```suggestion
[component dependencies]: ./ch16-02-02-component-dependencies.md
```

File: src/ch16-03-upgradeability.md [hunk 5]
```diff
@@ -28,10 +31,12 @@ To upgrade a contract, expose an entry point that executes `replace_class_syscal
 OpenZeppelin Contracts for Cairo provides the `Upgradeable` component that can be embedded into your contract to make it upgradeable. This component is a simple way to add upgradeability to your contract while relying on an audited library. It can be combined with the `Ownable` component to restrict the upgradeability to a single address, so that the contract owner has the exclusive right to upgrade the contract.
 
 ```rust,noplayground
-{{#include ../listings/ch99-starknet-smart-contracts/listing_99_12_upgradeable_contract_example/src/lib.cairo}}
+{{#include ../listings/ch16-building-advanced-starknet-smart-contracts/listing_07_oz_upgrade/src/lib.cairo}}
```
Comment (enitrat) [inline src/ch16-03-upgradeability.md:34]: how was this still passing? 🤔

Comment (TAdev0) [reply src/ch16-03-upgradeability.md:34]: listing_99 ... actually only contained a src folder with a lib.cairo file, nothing else, no configuration file. Maybe thats why?

**Comment by TAdev0** [comment]
@enitrat  ready to merge

File: src/ch16-03-upgradeability.md (hunk 5 above)
Comment (enitrat) [reply src/ch16-03-upgradeability.md:34]: there should be e file to delete then no ?

Comment (TAdev0) [reply src/ch16-03-upgradeability.md:34]: no the fil has been moved to a new listing, and i deleted the old folder that contained only a lib.cairo file

**Review by enitrat — APPROVED** [review]

**Pull request merged**